### PR TITLE
Check tar type before any other type

### PR DESCRIPTION
**Pull request by AshtonRoute**
Fixes sindresorhus/file-type#241

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,2 +1,4 @@
 node_modules
 yarn.lock
+.idea
```
Comment (kevva) [inline .gitignore]: These are unrelated. Please remove.

**Comment by AshtonRoute** [comment]
@kevva 

>Could you add a test for this as well?

I did.

https://github.com/sindresorhus/file-type/raw/2a9d3c26d7da81623809718b17e58a0150e5f5cf/fixture/fixture.tar

**Comment by kevva** [comment]
> I did.

Sorry, my bad. Did not spot you changing the fixture file :).

File: .gitignore (hunk 1 above)
Comment (AshtonRoute) [reply .gitignore]: Done

**Comment by AshtonRoute** [comment]
Another consideration:
```javascript
if (
    check([0x30, 0x30, 0x30, 0x30, 0x30, 0x30], {offset: 148, mask: [0xF8, 0xF8, 0xF8, 0xF8, 0xF8, 0xF8]}) && // Valid tar checksum
    tarHeaderChecksumMatches(buffer)
) {
    return {
        ext: 'tar',
        mime: 'application/x-tar'
    };
}
```

Right now this function returns 'tar' only if both Magic Number and the checksum are valid.

I suggest that we check for Magic Number first
and then if the checksum is valid return 'tar'
Otherwise throw an error (breaking change) or return null

File: index.js [hunk 2]
```diff
@@ -46,6 +46,7 @@ const fileType = input => {
 
 	const checkString = (header, options) => check(stringToBytes(header), options);
 
+	// we have to check for tar type before any other type
```
Comment (kevva) [inline index.js]: I'd either clarify why or remove this comment altogether.

**Comment by sindresorhus** [comment]
@AshtonRoute Agreed. Make it:

```js
if (
	check([0x30, 0x30, 0x30, 0x30, 0x30, 0x30], {offset: 148, mask: [0xF8, 0xF8, 0xF8, 0xF8, 0xF8, 0xF8]})
) {
	if (!tarHeaderChecksumMatches(buffer)) {
		return;
	}

	return {
		ext: 'tar',
		mime: 'application/x-tar'
	};
}
```

**Comment by sindresorhus** [comment]
And don't change the existing fixture, but instead add another one.

**Comment by sindresorhus** [comment]
@AshtonRoute Bump
